src/transformers/utils/import_utils.py

```diff
@@ -57,12 +57,13 @@ def _is_package_available(pkg_name: str, return_version: bool = False) -> Union[
                         package_version = temp_version
                         package_exists = True
                     else:
-                        package_exists = False
+                        package_version = "N/A"
                 except ImportError:
                     # If the package can't be imported, it's not available
-                    package_exists = False
-            else:
-                # For packages other than "torch", don't attempt the fallback and set as not available
+                    package_version = "N/A"
```

Comment:
At this line, the torch can't be imported, no? So why we can't keep `package_exists = False`?

Reply:
`torch` can be imported at this line! We only get here if `importlib.util.find_spec("torch")` is True, right?